### PR TITLE
converting -0 to 0 when it appears

### PR DESCRIPTION
**Pull request by LucasHFS**
Attempt to solve #2555 .
handled the possibility of getting result -0 in all uses of Math.ceil and converted it in 0

### CODE REVIEW

File: src/differenceInSeconds/index.ts [hunk 1]
```diff
@@ -34,5 +35,5 @@ export default function differenceInSeconds(
   requiredArgs(2, arguments)
 
   const diff = differenceInMilliseconds(dirtyDateLeft, dirtyDateRight) / 1000
-  return diff > 0 ? Math.floor(diff) : Math.ceil(diff)
-}
+   return removesNegativeZeroIfPresent(diff > 0 ? Math.floor(diff) : Math.ceil(diff))
+}
```
Comment (wenderjean) [inline src/differenceInSeconds/index.ts:39]: You a new-line problem here.

File: src/utils/removesNegativeZeroIfPresent/index.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export default function removesNegativeZeroIfPnumberent(number:number) : number {
+    return Object.is(number, -0) ? 0 : number
```
Comment (wenderjean) [inline src/utils/removesNegativeZeroIfPresent/index.ts:2]: The indentation does not follow the project pattern for those files you created.

File: src/differenceInHours/index.ts [hunk 3]
```diff
@@ -34,5 +35,5 @@ export default function differenceInHours(dirtyDateLeft: Date | number, dirtyDat
   const diff =
     differenceInMilliseconds(dirtyDateLeft, dirtyDateRight) /
     MILLISECONDS_IN_HOUR
-  return diff > 0 ? Math.floor(diff) : Math.ceil(diff)
+  return removesNegativeZeroIfPresent(diff > 0 ? Math.floor(diff) : Math.ceil(diff))
```
Comment (wenderjean) [inline src/differenceInHours/index.ts:38]: The same pattern repeats for all those pieces of code using `removesNegativeZeroIfPresent`, I mean, `diff > 0 ? Math.floor(diff) : Math.ceil(diff)`. What do u think about moving it to the utility function itself?

File: src/differenceInSeconds/index.ts (hunk 1 above)
Comment (LucasHFS) [reply src/differenceInSeconds/index.ts]: solved!

File: src/utils/removesNegativeZeroIfPresent/index.ts (hunk 2 above)
Comment (LucasHFS) [reply src/utils/removesNegativeZeroIfPresent/index.ts]: fixed

File: src/differenceInHours/index.ts (hunk 3 above)
Comment (LucasHFS) [reply src/differenceInHours/index.ts]: I agree that's a better way to approach it. I did it in the last commit.

**Comment by pabrodez** [comment]
Should date-fns still give the choice of how to round a value? #2511

**Comment by tan75** [comment]
> After this PR, should date-fns still give the choice of how to round a value? #2511

Good point! 
We've discussed this PR internally and have a couple of ideas - I will need to make some changes to these PRs.
The response to your question is 'Yes' - a rounding method will be passed in as an option with `Math.floor` as a default value.

File: src/_lib/getRoundingFn/index.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { RoundingMethod } from '../../types'
+
+export default function getRoundingFn(method: RoundingMethod) {
+  return {
+    ceil: Math.ceil,
+    round: Math.round,
+    floor: Math.floor,
+  }[method]
```
Comment (kossnocorp) [inline src/_lib/getRoundingFn/index.ts:8]: @tan75 please extract the object outside the function. Also, `trunc` method (which is the default in date-fns) is missing.

File: src/differenceInHours/index.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import differenceInMilliseconds from '../differenceInMilliseconds/index'
 import requiredArgs from '../_lib/requiredArgs/index'
-
-const MILLISECONDS_IN_HOUR = 3600000
+import getRoundedValue from '../utils/getRoundedValue'
```
Comment (kossnocorp) [inline src/differenceInHours/index.ts:3]:
```suggestion
import getRoundedValue from '../utils/getRoundedValue/index'
```

File: src/_lib/getRoundingFn/index.ts [hunk 6]
```diff
@@ -0,0 +1,9 @@
+import { RoundingMethod } from '../../types'
+
+export default function getRoundingFn(method: RoundingMethod) {
```
Comment (kossnocorp) [inline src/_lib/getRoundingFn/index.ts:3]: Please set the return type `: (input: number) => number`

**Review by kossnocorp — CHANGES_REQUESTED** [review]

File: src/differenceInWeeks/test.ts [hunk 7]
```diff
@@ -1,59 +1,103 @@
...
     )
     assert(result === 1)
   })
 
-  it('returns a negative number if the time value of the first date is smaller', function () {
+  it('returns the number of weeks between the given dates with `trunc` passed in as a rounding method', () => {
+    const result = differenceInWeeks(
+      new Date(2014, 6 /* Jul */, 8, 18, 0),
+      new Date(2014, 5 /* Jun */, 29, 6, 0),
+      'ceil'
```
Comment (fturmel) [inline src/differenceInWeeks/test.ts:19]: Shouldn't this argument be either 'trunc' or removed, and the assertion === 1?

Comment (tan75) [reply src/differenceInWeeks/test.ts:19]: @fturmel 
yes, you are right! my bad!

Comment (tan75) [reply src/differenceInWeeks/test.ts]: ✅  fixed

File: src/_lib/roundingMethods/index.ts [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+type RoundingFn = typeof Math.round
+
+type RoundingFnsMap = { [method in RoundingMethod]: RoundingFn }
+
+const RoundingMap: RoundingFnsMap = {
+  ceil: Math.ceil,
+  round: Math.round,
+  floor: Math.floor,
+  trunc: (value: number) => parseInt(value.toString()), // Math.trunc is not supported by IE
```
Comment (kossnocorp) [inline src/_lib/roundingMethods/index.ts:11]:
```suggestion
  trunc: (value: number) => value < 0 ? Math.ceil(value) : Math.floor(value), // Math.trunc is not supported by IE
```

[Benchmark](https://perf.link/#eyJpZCI6ImtoZnN0ZThnbzlyIiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImZ1bmN0aW9uIG1hdGhUcnVuYyAodmFsdWUpIHsgcmV0dXJuIHZhbHVlIDwgMCA%2FIE1hdGguY2VpbCh2YWx1ZSkgOiBNYXRoLmZsb29yKHZhbHVlKSB9XG5mdW5jdGlvbiBwYXJzZVRydW5jKHZhbHVlKSB7IHJldHVybiBwYXJzZUludCh2YWx1ZS50b1N0cmluZygpKSB9IiwidGVzdHMiOlt7Im5hbWUiOiJGaW5kIGl0ZW0gMTAwIiwiY29kZSI6Im1hdGhUcnVuYygtMS45KSIsInJ1bnMiOlsyMjAwMDAsMTQ2NTAwMCw0OTUwMDAsMjI2MTAwMCw0MTMyMDAwLDI5MjAwMCw5MDMwMDAsMzE2OTAwMCwxMzAwMCwyNDk3MDAwLDIyMDAwLDEzMDAwLDcxMTAwMCwyMzAwMCwyNDc4MDAwLDIyMDAwLDEzMDAwLDIwNTgwMDAsOTM1MDAwLDIyMjgwMDAsMjQyMDAwMCwxMDkwMDAwLDEzNDMwMDAsOTA0MDAwLDE3NTEwMDAsMTUwMDAwLDIyODMwMDAsMTAwMCw0NjU2MDAwLDIxNjgwMDAsNTg4MDAwLDc5MzAwMCwyMTQyMDAwLDMzMjAwMDAsMjM0MDAwLDE1ODAwMDAsMTUzMTAwMCwxMzAwMCwzMTg4MDAwLDEzMDAwLDEzMDAwLDIzMjYwMDAsMzI2OTAwMCwxNjE4MDAwLDMxNjAwMDAsMTI0NzAwMCwzNDEyMDAwLDQyMzAwMDAsMTMwMDAsMzc4NjAwMCwyNDcxMDAwLDMyNzcwMDAsMjM1NjAwMCwxNzMwMDAsOTQ1MDAwLDMzMTkwMDAsMjM3ODAwMCwyNTIwMDAsMTMwMDAsMzMxNTAwMCwxMzAwMCwxMzAwMCw0NDY2MDAwLDE2NjgwMDAsMTAwMCwxMzAwMCw0MzMxMDAwLDE3NjgwMDAsMzM0MDAwLDEzMDAwLDEwNTkwMDAsMTMwMDAsMjIwMDAsMjE5OTAwMCw2ODAwMDAsNDY3MDAwLDU1MzAwMCw0MzA0MDAwLDE0MDAwLDIyMDAwLDI5OTAwMCw2NTkwMDAsOTk1MDAwLDIyMDAwLDMzMjIwMDAsMTczMzAwMCwxNzk2MDAwLDMyODUwMDAsMjU4ODAwMCwzNTY3MDAwLDQwNjcwMDAsNTYwMDAwLDI3MDAwLDIzNzAwMCwzMjMzMDAwLDMyMTQwMDAsMTAwMCwzMDcwMDAsMzM4NTAwMCwyMDMwMDAwXSwib3BzIjoxNTI5NzEwfSx7Im5hbWUiOiJGaW5kIGl0ZW0gMjAwIiwiY29kZSI6Ik1hdGgudHJ1bmMoLTEuOSkiLCJydW5zIjpbMzIxODAwMCwyMTAwMCwxODAwMCw0OTcwMDAsMjEwMDAsOTAwMDAwLDc2MzAwMCwzMjcwMDAsNDkwMDAwLDEyODMwMDAsMjU4MjAwMCw0ODE4MDAwLDI2NTIwMDAsNzc2MDAwLDk4NjAwMCwyMTAwMCwxODE0MDAwLDI5NjEwMDAsNDM3NzAwMCwyNzU2MDAwLDEwMDAsMjEwMDAsMTU0MDAwMCwxMzIzMDAwLDEyMTMwMDAsMTMzMDAwLDI3OTIwMDAsMjEwMDAsNzkwMDAwLDU2OTAwMCw4MTcwMDAsMTg0MDAwMCwyMTAwMCwxNTIwMDAsMTU3MzAwMCwxMzc1MDAwLDIwMTUwMDAsNTg3MDAwLDM0OTkwMDAsMTE1ODAwMCw0NjMzMDAwLDMyMTcwMDAsNDU1MDAwMCwzODE0MDAwLDI0MDAwLDU5MDAwMCwxNTQwMDAsMjEwMDAsNDA0MjAwMCwyMTAwMCwxOTgwMDAsODEwMDAwLDM2MjAwMCwyNTEwMDAsNzYzMDAwLDMxMTAwMCwyMzgzMDAwLDM4MzUwMDAsMjE4MjAwMCwyMTAwMCwyMTgwMDAwLDEzMzAwMCwyMTAwMCwzNzMwMDAsNjc2MDAwLDgxMDAwMCwzNjA2MDAwLDIxMDAwLDUwNzAwMCw2OTAwMDAsMTkwODAwMCwxODU5MDAwLDM0MzEwMDAsODkwMDAwLDEyODMwMDAsMzMzMDAwMCw0MjQ5MDAwLDQxMTQwMDAsMjEwMDAsMjgwNDAwMCwyMjA0MDAwLDIxMDAwLDIxMDAwLDQzNjIwMDAsMjEwMDAsMjAwNjAwMCwxNTkyMDAwLDM5MTQwMDAsMjAxMzAwMCwyODI3MDAwLDEwMTEwMDAsMjc3MzAwMCwyNzYxMDAwLDIxMDAwLDE5OTkwMDAsMTIwOTAwMCwxNTUwMDAsMTMxNDAwMCwzODY1MDAwLDExMzIwMDBdLCJvcHMiOjE1MzAzMDB9LHsibmFtZSI6IkZpbmQgaXRlbSA0MDAiLCJjb2RlIjoicGFyc2VUcnVuYygtMS45KSIsInJ1bnMiOlsyMDQyMDAwLDkwMDAsMTY0ODAwMCwyNzQ5MDAwLDIyMDAwLDY5NTAwMCw5MDAwLDEzMTQwMDAsNTY3MDAwLDQ4MzAwMCw5MDAwLDEzNzAwMCw0MDAwMDAsMTUwMjAwMCw3NzUwMDAsMjU1NDAwMCwzMTAxMDAwLDIyMDAwLDIzNzIwMDAsMjAyNDAwMCwxMDAwLDE4MTQwMDAsMTAxMDAwMCw2NDQwMDAsMTgyMDAwMCwzMDEwMDAsMjIxNTAwMCw0ODEwMDAsMTU4NDAwMCwyMTMwMDAwLDIyNjMwMDAsNTA3MDAwLDc2MjAwMCwzMDQ0MDAwLDk3NTAwMCw5MDAwLDEzNjIwMDAsMTczNzAwMCw4NjcwMDAsMTU5NzAwMCwyMjY1MDAwLDgyODAwMCw3MjQwMDAsMzYyMDAwLDIwMDAsODkwMDAsOTAwMCwyNDU3MDAwLDExMTkwMDAsMTc5MzAwMCwxMzUxMDAwLDkwMDAsMjIwMDAsMTAwMDAsMTM2MDAwLDEwNDgwMDAsNDQyMDAwLDI3NDEwMDAsMjAwODAwMCw1MjYwMDAsMTc3NTAwMCwzNTEwMDAsMTgyMzAwMCw5MDAwLDI3NTMwMDAsNDQ3MDAwLDg4MzAwMCwyNzY1MDAwLDI1NTcwMDAsMTY2MDAwMCw4NzAwMDAsNTA5MDAwLDEyNzAwMCw5MDAwLDE3OTQwMDAsMzI0MDAwLDM2MjAwMCwxNTYwMDAwLDE4NTIwMDAsNTIwMDAwLDE2MTUwMDAsNTAxMDAwLDkwMDAsMjA2NDAwMCw0ODQwMDAsNzAyMDAwLDI0MDAwLDIxMDAwLDIzNjUwMDAsMjIwMDAsMzA3MDAwMCwyMjAwMCw2NzkwMDAsMjE2NjAwMCwyNDkzMDAwLDE3NzUwMDAsMTM3OTAwMCwyMjI2MDAwLDIyMDAwLDIyMjAwMF0sIm9wcyI6MTEwMzA5MH1dLCJ1cGRhdGVkIjoiMjAyMS0wOS0wN1QwNDo0MzozOC42NThaIn0%3D)

File: src/_lib/roundingMethods/index.ts [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+}
+
+export function getRoundingMethod(method: RoundingMethod) {
+  return RoundingMap[method]
+}
+
+export const defaultRoundingMethod = 'trunc'
```
Comment (kossnocorp) [inline src/_lib/roundingMethods/index.ts:18]:
```suggestion
export const defaultRoundingMethod : RoundingMethod = 'trunc'
```

**Review by kossnocorp — CHANGES_REQUESTED** [review]

File: src/differenceInHours/index.ts [hunk 10]
```diff
@@ -17,22 +21,27 @@ const MILLISECONDS_IN_HOUR = 3600000
  *
  * @param {Date|Number} dateLeft - the later date
  * @param {Date|Number} dateRight - the earlier date
+ * @param {RoundingMethod} roundingMethod - the rounding method (`ceil`, `floor`, `round` or `trunc`) with `trunc` by default.
```
Comment (kossnocorp) [inline src/differenceInHours/index.ts:24]:
```suggestion
 * @param {String} roundingMethod - the rounding method (`ceil`, `floor`, `round` or `trunc`) with `trunc` by default.
```

File: src/differenceInMinutes/index.ts [hunk 11]
```diff
@@ -17,30 +21,35 @@ var MILLISECONDS_IN_MINUTE = 60000
  *
  * @param {Date|Number} dateLeft - the later date
  * @param {Date|Number} dateRight - the earlier date
+ * @param {RoundingMethod} roundingMethod - the rounding method (`ceil`, `floor`, `round` or `trunc`) with `trunc` by default.
```
Comment (kossnocorp) [inline src/differenceInMinutes/index.ts:24]:
```suggestion
 * @param {String} roundingMethod - the rounding method (`ceil`, `floor`, `round` or `trunc`) with `trunc` by default.
```

File: src/differenceInQuarters/index.ts [hunk 12]
```diff
@@ -1,34 +1,42 @@
...
- * Get the number of full quarters between the given dates.
+ * Get the number of quarters between the given dates.
  *
  * ### v2.0.0 breaking changes:
  *
  * - [Changes that are common for the whole library](https://github.com/date-fns/date-fns/blob/master/docs/upgradeGuide.md#Common-Changes).
  *
  * @param {Date|Number} dateLeft - the later date
  * @param {Date|Number} dateRight - the earlier date
+ * @param {RoundingMethod} roundingMethod - the rounding method (`ceil`, `floor`, `round` or `trunc`) with `trunc` by default.
```
Comment (kossnocorp) [inline src/differenceInQuarters/index.ts:23]:
```suggestion
 * @param {String} roundingMethod - the rounding method (`ceil`, `floor`, `round` or `trunc`) with `trunc` by default.
```

File: src/differenceInSeconds/index.ts [hunk 13]
```diff
@@ -15,6 +20,7 @@ import requiredArgs from '../_lib/requiredArgs/index'
  *
  * @param {Date|Number} dateLeft - the later date
  * @param {Date|Number} dateRight - the earlier date
+ * @param {RoundingMethod} roundingMethod - the rounding method (`ceil`, `floor`, `round` or `trunc`) with `trunc` by default.
```
Comment (kossnocorp) [inline src/differenceInSeconds/index.ts:23]:
```suggestion
 * @param {String} roundingMethod - the rounding method (`ceil`, `floor`, `round` or `trunc`) with `trunc` by default.
```

File: src/differenceInWeeks/index.ts [hunk 14]
```diff
@@ -24,6 +29,7 @@ import requiredArgs from '../_lib/requiredArgs/index'
  *
  * @param {Date|Number} dateLeft - the later date
  * @param {Date|Number} dateRight - the earlier date
+ * @param {RoundingMethod} roundingMethod - the rounding method (`ceil`, `floor`, `round` or `trunc`) with `trunc` by default.
```
Comment (kossnocorp) [inline src/differenceInWeeks/index.ts:32]:
```suggestion
 * @param {String} roundingMethod - the rounding method (`ceil`, `floor`, `round` or `trunc`) with `trunc` by default.
```

**Review by kossnocorp — APPROVED** [review]

File: src/differenceInQuarters/index.ts [hunk 15]
```diff
@@ -1,34 +1,42 @@
+import type { RoundingMethod } from 'src/types'
```
Comment (kossnocorp) [inline src/differenceInQuarters/index.ts:1]:
```suggestion
import type { RoundingMethod } from '../types'
```

File: src/differenceInMinutes/index.ts [hunk 16]
```diff
@@ -1,7 +1,11 @@
+import type { RoundingMethod } from 'src/types'
```
Comment (kossnocorp) [inline src/differenceInMinutes/index.ts:1]:
```suggestion
import type { RoundingMethod } from '../types'
```

File: src/differenceInSeconds/index.ts [hunk 17]
```diff
@@ -1,5 +1,10 @@
+import type { RoundingMethod } from 'src/types'
```
Comment (kossnocorp) [inline src/differenceInSeconds/index.ts:1]:
```suggestion
import type { RoundingMethod } from '../types'
```

Comment (kossnocorp) [inline src/differenceInSeconds/index.ts]:
```suggestion
import type { RoundingMethod } from '../types'
```

File: src/differenceInWeeks/index.ts [hunk 18]
```diff
@@ -1,5 +1,10 @@
+import type { RoundingMethod } from 'src/types'
```
Comment (kossnocorp) [inline src/differenceInWeeks/index.ts:1]:
```suggestion
import type { RoundingMethod } from '../types'
```

File: src/differenceInHours/index.ts [hunk 19]
```diff
@@ -1,7 +1,11 @@
+import type { RoundingMethod } from '../../src/types'
```
Comment (kossnocorp) [inline src/differenceInHours/index.ts:1]:
```suggestion
import type { RoundingMethod } from '../types'
```

Comment (kossnocorp) [inline src/differenceInHours/index.ts]:
```suggestion
import type { RoundingMethod } from '../types'
```

**Review by tan75 — APPROVED** [review]

**Pull request merged**